libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs

```diff
@@ -194,16 +194,27 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             // Single command running with a copy of the original data
             client.DefaultRequestHeaders.Clear();
 
-            JToken instanceBody = null;
+            object instanceBody = null;
+            string instanceBodyErr = null;
+
             if (this.Body != null)
             {
-                var (body, err) = this.Body.TryGetValue(dc.State);
-                if (err != null)
+                if (this.Body.Value is JObject)
                 {
-                    throw new ArgumentException(err);
+                    // For raw JSON Object, replace the jtoken recursively 
+                    var jObjectBody = (JToken)JToken.FromObject(this.Body.Value).DeepClone();
```

Comment:
I'd remove the cast to `JToken`. `JToken.FromObject` returns a `JToken`, so there should be no type ambiguity.